src/main/ngapp/src/app/dataservices/shared/dataservice.service.ts

```diff
@@ -358,40 +374,122 @@ export class DataserviceService extends ApiService {
   }
 
   /**
-   * Export a dataservice to a git repository
+   * Converts a base64 data string into a blob for use with the FileSaver library
+   * Acknowledgement to
+   * http://stackoverflow.com/questions/16245767/creating-a-blob-from-a-base64-string-in-javascript
+   */
+  private b64toBlob(b64Data: string, contentType: string): Blob {
+    contentType = contentType || '';
+    let sliceSize = 512;
+
+    //
+    // Decodes the base64 string back into binary data byte characters
+    //
+    let byteCharacters = atob(b64Data);
+    let byteArrays = [];
+
+    //
+    // Each character's code point (charCode) will be the value of the byte.
+    // Can create an array of byte values by applying this using the .charCodeAt
+    // method for each character in the string.
+    //
+    // The performance can be improved a little by processing the byteCharacters
+    // in smaller slices, rather than all at once. Rough testing indicates 512 bytes
+    // seems to be a good slice size.
+    //
+    for (let offset = 0; offset < byteCharacters.length; offset += sliceSize) {
+        let slice = byteCharacters.slice(offset, offset + sliceSize);
+
+        let byteNumbers = new Array(slice.length);
+        for (let i = 0; i < slice.length; i++) {
+            byteNumbers[i] = slice.charCodeAt(i);
+        }
+
+        //
+        // Convert the array of byte values into a real typed byte array
+        // by passing it to the Uint8Array constructor.
+        //
+        let byteArray = new Uint8Array(byteNumbers);
+        byteArrays.push(byteArray);
+    }
+
+    //
+    // Convert to a Blob by wrapping it in an array passing it to the Blob constructor.
+    //
+    let blob = new Blob(byteArrays, {
+        type: contentType
+    });
+
+    return blob;
+  }
+
+  /**
+   * Download a dataservice as a jar archive
    * @param {string} dataserviceName the dataservice name
    * @returns {Observable<boolean>}
    */
-  public exportDataservice(dataserviceName: string): Observable<boolean> {
-    const repoPathKey = this.appSettings.GIT_REPO_PATH_KEY;
-    const repoBranchKey = this.appSettings.GIT_REPO_BRANCH_KEY;
-    const repoUsernameKey = this.appSettings.GIT_REPO_USERNAME_KEY;
-    const repoPasswordKey = this.appSettings.GIT_REPO_PASSWORD_KEY;
-    const repoAuthorEmailKey = this.appSettings.GIT_REPO_AUTHOR_EMAIL_KEY;
-    const repoAuthorNameKey = this.appSettings.GIT_REPO_AUTHOR_NAME_KEY;
-    const repoFilePathKey = this.appSettings.GIT_REPO_FILE_PATH_KEY;
-
+  public downloadDataservice(dataserviceName: string): Observable<boolean> {
```

Comment:
Would be nice if this could also be implemented in `MockDataserviceService` so that the non-OpenShift dev mode would work.